### PR TITLE
Analytics: Set user property for permissions and remove file fetched

### PR DESCRIPTION
**Pull request by KattMingMing**
##### Description

- File fetched isn't providing any value now that backend tracking is implemented.
- We should be explicitly setting a user property for github permissions to reduce confusion when querying

### CODE REVIEW

File: app/web_modules/sourcegraph/user/UserActions.js [hunk 1]
```diff
@@ -61,10 +61,12 @@ export class FetchedEmails {
 export class FetchedGitHubToken {
 	uid: number;
 	token: ExternalToken | {Error: any};
+	eventName: string;
 
 	constructor(uid: number, token: ExternalToken | {Error: any}) {
 		this.uid = uid;
 		this.token = token;
+		this.eventName = "FetchedGitHubToken";
```
Comment (uforic) [inline app/web_modules/sourcegraph/user/UserActions.js]: Do we have a string constants file for this type of thing nowadays?

File: app/web_modules/sourcegraph/util/EventLogger.js [hunk 2]
```diff
@@ -258,6 +258,12 @@ export class EventLogger {
 				this.logEventForCategory(AnalyticsConstants.CATEGORY_ENGAGEMENT, AnalyticsConstants.ACTION_SUCCESS, action.eventName);
 			}
 			break;
+		case UserActions.FetchedGitHubToken:
+			if (action.eventName && action.token) {
+				let allowedPrivateAuth = action.token.scope && action.token.scope.includes("repo") && action.token.scope.includes("read:org");
```
Comment (uforic) [inline app/web_modules/sourcegraph/util/EventLogger.js]: is eventName used here / why do we care about event name?

Comment (uforic) [inline app/web_modules/sourcegraph/util/EventLogger.js]: Also, just confirming we are not logging this - this is because it is done server side?

**Comment by uforic** [comment]
LGTM - Just confirm that FetchedGitHubToken doesn't log, and that having an action eventName is necessary.

File: app/web_modules/sourcegraph/util/EventLogger.js (hunk 2 above)
Comment (KattMingMing) [inline app/web_modules/sourcegraph/util/EventLogger.js]: We need an explicit property set on the user in amplitude land - this is not currently done. We can remove event name from this

**Comment by KattMingMing** [comment]
FetchedGitHubToken won't be recorded since there is no more eventNames and it is also inside of it's own case

**Pull request merged**